src/signatures/winternitz.rs

```diff
@@ -133,7 +133,11 @@ pub fn sign_digits(secret_key: &str, message_digits: [u8; N0 as usize]) -> Vec<D
 
     let mut signatures = Vec::new();
     for i in 0..N {
-        let digit_signature = digit_signature(secret_key, i, checksum_digits[(N - 1 - i) as usize]);
+        let digit_signature = digit_signature(
+            secret_key,
+            i.try_into().unwrap(),
```

Comment:
Can we cast `i` to u32 instead of doing `try_into().unwrap()`?
```suggestion
            i as u32,
```